File: crud-item-list/src/main/java/com/gurunars/crud_item_list/PositionFetcher.java
package com.gurunars.crud_item_list;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Set;

import java8.util.function.BiFunction;

class PositionFetcher<ItemType> implements BiFunction<List<ItemType>, Set<ItemType>, List<Integer>> {

    @Override
    public List<Integer> apply(List<ItemType> all, Set<ItemType> selectedItems) {
        List<Integer> positions = new ArrayList<>();

        if (all.isEmpty() || selectedItems.isEmpty()) {
            return positions;
        }

        for (ItemType cursor: selectedItems) {
            int index = all.indexOf(cursor);
            if (index != -1) {
                positions.add(index);
            }
        }
        Collections.sort(positions);
        return positions;
    }
}

File: crud-item-list/src/main/kotlin/com/gurunars/crud_item_list/PositionFetcher.kt
package com.gurunars.crud_item_list

import java.util.ArrayList
import java.util.Collections

import java8.util.function.BiFunction

internal class PositionFetcher<ItemType> : BiFunction<List<ItemType>, Set<ItemType>, List<Int>> {

    override fun apply(all: List<ItemType>, selectedItems: Set<ItemType>): List<Int> {
        val positions = ArrayList<Int>()

        if (all.isEmpty() || selectedItems.isEmpty()) {
            return positions
        }

        for (cursor in selectedItems) {
            val index = all.indexOf(cursor)
            if (index != -1) {
                positions.add(index)
            }
        }
        Collections.sort(positions)
        return positions
    }
}
